rpc/connection_test.go

```diff
@@ -256,6 +256,20 @@ func TestConnectionClientCallError(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestConnectionClientCallCompressedError(t *testing.T) {
+	serverConn, conn := MakeConnectionForTest(t)
+	defer conn.Shutdown()
+
+	c := connectionClient{conn}
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- c.CallCompressed(context.Background(), "callRpc", nil, nil, CompressionGzip)
+	}()
+	serverConn.Close()
+	err := <-errCh
+	require.Error(t, err)
+}
```

Comment:
these are tested in the `protocol_test.go` i'm not sure the setup exactly to get a happy path working (i think need more mock layers here) but i'm not sure it's worth it since there is coverage there already